strands_human_help/src/strands_human_help/help_screen.py

```diff
@@ -76,19 +68,34 @@ def being_helped(self, failed_component, interaction_service, n_fails):
         self.generate_help_content(self.finish_label, self.help_instructions_html,  interaction_service)
 
     def help_finished(self, failed_component, interaction_service, n_fails):
-        self.display_main_page()
+        self.remove_help_content()
 
     def help_failed(self, failed_component, interaction_service, n_fails):
-        self.ask_help(failed_component, interaction_service, n_fails)
-
+        self.remove_help_content()
+        
+    def remove_help_content(self):
+        self.screen.cancel_all_goals()
 
-    def display_main_page(self,):
-        strands_webserver.client_utils.display_relative_page(self.display_no, 'index.html')
 
-    def generate_help_content(self,label, html, interaction_service):
+    def generate_help_content(self, label, html, interaction_service):
+        interaction_success = rospy.ServiceProxy("/monitored_navigation/" + interaction_service, Empty)
         if label is None:
-            content=self.magnetic_stip_help_html
+            goal = ModalDialogSrvGoal(title = "Please help",
+                                  text = html,
+                                  buttons = [],
+                                  buttons_class = []
+                                  )
         else:
-            buttons = [(label, interaction_service)]
-            content = strands_webserver.page_utils.generate_alert_button_page(html, buttons, self.service_prefix)
-        strands_webserver.client_utils.display_content(self.display_no, content)
+            goal = ModalDialogSrvGoal(title = "Please help",
+                                  text = html,
+                                  buttons = [label],
+                                  buttons_class = ['btn-success']
+                                  )
+        self.screen.send_goal(goal)
+        self.screen.wait_for_result()
```

Comment:
yes, another thread does the preemption when the timeout occurs, and this wait is released when that happens. The result is an empty msg in that case, that's why i then check if the button was pressed below